website/specs/cdsc-wg1-02/overview.md

```diff
@@ -8,10 +8,515 @@ meta_description: Linux Foundation Energy - Carbon Data Specification Consortium
+            "documentation": "https://demoutility.com/docs/oauth/registration#sou",
+            "format": "string_or_null",
+            "max_length": 500,
+            "default": null,
+        },
+        "tos_url": {
+            "id": "tos_url",
+            "type": "registration_field",
+            "field_name": "cds_initial_scope_of_use_tos_url",
+            "description": "Optionally, provide a link to your terms for your scope of use.",
+            "documentation": "https://demoutility.com/docs/oauth/registration#tos_url",
+            "format": "url_or_null",
+            "default": null,
+        },
+        "dir_autolist": {
+            "id": "dir_autolist",
+            "type": "registration_field",
+            "field_name": "cds_autolist_in_directory",
+            "description": "Do you want to be automatically listed in the public client directory when registered and approved?",
+            "documentation": "https://demoutility.com/docs/oauth/registration#dir-autolist",
+            "format": "boolean",
+            "default": true,
+        },
+        "review_step1": {
+            "id": "review_step1",
+            "type": "manual_review",
+            "name": "Utility Review",
+            "description": "Demo Utility will manually review your client registration before enabling live access.",
+            "documentation": "https://demoutility.com/docs/oauth/registration#review-process",
+        },
+        "setup_fee1": {
+            "id": "setup_fee1",
+            "type": "payment_required",
+            "name": "Registration Fee",
+            "amount": 2000,
+            "currency": "USD",
+            "description": "Demo Utility requires a setup fee payment of $20 before enabling live access.",
+            "documentation": "https://demoutility.com/docs/oauth/registration#review-process",
+        },
+        ...
+    ],
+}
+```
+
+## Dynamic Client Registration <a id="client-registration" href="#client-registration" class="permalink">🔗</a>
+
+```
+==Request==
+POST /.well-known/carbon-data-spec.json HTTP/1.1
+Content-Type: application/json
+Host: demoutility.com
+
+{
+    # Normal OAuth registration fields
+    "client_name": "Example EV Company",
+    "client_uri": "https://example.com",
+    "logo_uri": "https://example.com/logo.png",
+    "tos_uri": "https://example.com/terms",
+    "policy_uri": "https://example.com/privacy",
+    "contacts": [
+        "mailto:aaa@bbb.com",
+        "tel:+15554443333",
+    ],
+    "scope": "client_admin accounts ..."
+    "redirect_uris": [
+        "https://client.example.org/callback",
+        "https://client.example.org/callback2",
+    ],
+
+    # CDS Extension - additional registration fields
+    "cds_initial_scope_of_use": "For testing purposes only.",
+    "cds_autolist_in_directory": false,
+}
+
+
+==Response==
+HTTP/1.1 201 Created
+Content-Type: application/json;charset=UTF-8
+
+{
+    ...client metadata entry (see client metadata endpoint for format)
```

Comment:
From @antarasargam ([on April 24, 2024](https://github.com/carbon-data-specification/Customer-Data/pull/25#discussion_r1578670991)):

---

If this is referring to the `Client Endpoint` example below, it makes sense to return the `client_secret` in the Registration Response but the example below does not work as a standalone call because the spec mentioned that the `client_secret` MUST not be returned for the Client API requests.

---